Zend/zend_vm_def.h

```diff
@@ -9380,6 +9380,130 @@ ZEND_VM_HANDLER(202, ZEND_CALLABLE_CONVERT, UNUSED, UNUSED)
+			property_val = OBJ_PROP(zobj, prop_offset);
+
+			if (Z_TYPE_P(property_val) != IS_UNDEF) {
+				zend_property_info *prop_info = (zend_property_info*) CACHED_PTR_EX(cache_slot + 2);
+
+				if (UNEXPECTED(prop_info != NULL)) {
+					if (UNEXPECTED(prop_info->flags & ZEND_ACC_READONLY)) {
+						Z_PROP_FLAG_P(property_val) |= IS_PROP_REINITABLE;
+					}
+					value = zend_assign_to_typed_prop(prop_info, property_val, value EXECUTE_DATA_CC);
+					if (UNEXPECTED(prop_info->flags & ZEND_ACC_READONLY)) {
+						Z_PROP_FLAG_P(property_val) &= ~IS_PROP_REINITABLE;
+					}
+					ZEND_VM_C_GOTO(free_and_exit_init_obj);
+				} else {
+ZEND_VM_C_LABEL(fast_init_obj):
+					value = zend_assign_to_variable(property_val, value, OP_DATA_TYPE, EX_USES_STRICT_TYPES());
+					if (UNEXPECTED(RETURN_VALUE_USED(opline))) {
+						ZVAL_COPY(EX_VAR(opline->result.var), value);
+					}
+					ZEND_VM_C_GOTO(exit_init_obj);
+				}
+			}
+		} else {
+			name = Z_STR_P(GET_OP2_ZVAL_PTR(BP_VAR_R));
+			if (EXPECTED(zobj->properties != NULL)) {
+				if (UNEXPECTED(GC_REFCOUNT(zobj->properties) > 1)) {
+					if (EXPECTED(!(GC_FLAGS(zobj->properties) & IS_ARRAY_IMMUTABLE))) {
+						GC_DELREF(zobj->properties);
+					}
+					zobj->properties = zend_array_dup(zobj->properties);
+				}
+				property_val = zend_hash_find_known_hash(zobj->properties, name);
+				if (property_val) {
+					ZEND_VM_C_GOTO(fast_init_obj);
+				}
+			}
+
+			if (!zobj->ce->__set && (zobj->ce->ce_flags & ZEND_ACC_ALLOW_DYNAMIC_PROPERTIES)) {
+				if (EXPECTED(zobj->properties == NULL)) {
+					rebuild_object_properties(zobj);
+				}
+				if (OP_DATA_TYPE == IS_CONST) {
+					if (UNEXPECTED(Z_OPT_REFCOUNTED_P(value))) {
+						Z_ADDREF_P(value);
+					}
+				} else if (OP_DATA_TYPE != IS_TMP_VAR) {
+					if (Z_ISREF_P(value)) {
+						if (OP_DATA_TYPE == IS_VAR) {
+							zend_reference *ref = Z_REF_P(value);
+							if (GC_DELREF(ref) == 0) {
+								ZVAL_COPY_VALUE(&tmp, Z_REFVAL_P(value));
+								efree_size(ref, sizeof(zend_reference));
+								value = &tmp;
+							} else {
+								value = Z_REFVAL_P(value);
+								Z_TRY_ADDREF_P(value);
+							}
+						} else {
+							value = Z_REFVAL_P(value);
+							Z_TRY_ADDREF_P(value);
+						}
+					} else if (OP_DATA_TYPE == IS_CV) {
+						Z_TRY_ADDREF_P(value);
+					}
+				}
+				zend_hash_add_new(zobj->properties, name, value);
+				if (UNEXPECTED(RETURN_VALUE_USED(opline))) {
+					ZVAL_COPY(EX_VAR(opline->result.var), value);
+				}
+				ZEND_VM_C_GOTO(exit_init_obj);
+			}
+		}
+	}
+
+	name = Z_STR_P(GET_OP2_ZVAL_PTR(BP_VAR_R));
+
+	if (OP_DATA_TYPE == IS_CV || OP_DATA_TYPE == IS_VAR) {
+		ZVAL_DEREF(value);
+	}
+
+	value = zobj->handlers->write_property(zobj, name, value, cache_slot);
+
+ZEND_VM_C_LABEL(free_and_exit_init_obj):
+	if (EXPECTED(RETURN_VALUE_USED(opline))) {
+		ZVAL_COPY_DEREF(EX_VAR(opline->result.var), object);
+	}
+	FREE_OP_DATA();
+ZEND_VM_C_LABEL(exit_init_obj):
+	FREE_OP2();
+	FREE_OP1();
```

Comment:
Seems correct for operand freeing other than my previous comment, as far as I can tell at a glance. I guess OP1 can be a reference for `clone (returns_reference()) with {...}`, so this cannot be combined with ZVAL_COPY_DEREF. There's also the obvious non-object cases that need to be freed

Reply:
Unfortuantely, I don't really understand this. What should I fix? Should I change the `ZVAL_COPY_DEREF` call to something else?

Reply:
Nothing needs to be done, I'm just making a note that it seems correct (since there's few people that understand this), and leaving a note to point to for why certain optimizations wouldn't actually work